File: Gameshop_Full/src/app/modelclasses.ts
import { Company, Game, Genre, User } from './domainclasses';

export class HomeModel {
    genres!: Genre[];
    companies!: Company[];
}

export class GameListModel {
    genres!: Genre[];
    companies!: Company[];
    games!: Game[];
}

export class LoginResult {
    user!: User;
    accessToken!: string;
    refreshToken!: string;
}

export class GameEditModel {
    genres!: Genre[];
    companies!: Company[];
    game!: Game;
}

File: Gameshop_Full/ModelClasses.cs
using Gameshop.Db;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Gameshop
{
	public class HomeModel
	{
		public List<Genre> Genres { get; set; }
		public List<Company> Companies { get; set; }
	}

	public class GameListModel
	{
		public List<Genre> Genres { get; set; }
		public List<Company> Companies { get; set; }
		public List<Game> Games { get; set; }
	}

	public class GameEditModel
	{
		public List<Genre> Genres { get; set; }
		public List<Company> Companies { get; set; }
		public Game Game { get; set; }
	}
}
